modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchDocumentationTest.java

```diff
@@ -0,0 +1,88 @@
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.elasticsearch.action.admin.cluster.health.ClusterHealthResponse;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.transport.TransportClient;
+import org.elasticsearch.cluster.health.ClusterHealthStatus;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.transport.TransportAddress;
+import org.elasticsearch.transport.client.PreBuiltTransportClient;
+import org.junit.Test;
+
+public class ElasticsearchDocumentationTest {
+
+    @Test
+    public void containerWithHttpClient() throws IOException {
+        // httpClientContainerStart {
+        // Create the elasticsearch container.
+        ElasticsearchContainer container = new ElasticsearchContainer("docker.elastic.co/elasticsearch/elasticsearch:6.4.1");
+
+        // Start the container. This step might take some time...
+        container.start();
+
+        // Do whatever you want with the rest client ...
+        final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+        credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials("elastic", "changeme"));
+        RestClient restClient = RestClient.builder(HttpHost.create(container.getHttpHostAddress()))
+            .setHttpClientConfigCallback(httpClientBuilder -> httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider))
+            .build();
+        Response response = restClient.performRequest(new Request("GET", "/"));
+        // }
+        // Compare Result
+        assertEquals(response.getStatusLine().getStatusCode(), 200);
+
+        // httpClientContainerStop {
+        // Stop the container.
+        container.stop();
+        // }
+    }
+
+    @Test
+    public void containerWithTransportClient() {
+        // Create the elasticsearch container.
+        ElasticsearchContainer container = new ElasticsearchContainer("docker.elastic.co/elasticsearch/elasticsearch:6.4.1");
+
+        // Start the container. This step might take some time...
+        container.start();
+
+        // transportClientContainerStart {
+        // ... or the transport client
+        TransportAddress transportAddress = new TransportAddress(container.getTcpHost());
+        Settings settings = Settings.builder().put("cluster.name", "docker-cluster").build();
+        TransportClient transportClient = new PreBuiltTransportClient(settings)
+            .addTransportAddress(transportAddress);
+        ClusterHealthResponse healths = transportClient.admin().cluster().prepareHealth().get();
+        // }
+        // Compare Result
+        assertEquals(ClusterHealthStatus.GREEN, healths.getStatus());
+
+        // Stop the container.
+        container.stop();
```

Comment:
please use try-with-resources to make sure that `stop()` is executed even if some assert fails

Reply:
done, but this changed the code in the documentation

Reply:
@raynigon 

1. please do not resolve conversations started by others.
2. Judging by other codeincludes, it _is_ possible to do it without changing the semantics in the docs. Please have a look at other codeincludes for some inspiration.

Reply:
First, sorry didnt know that.

Second, what i meant is, previously the docs showed container stopping with stop:
<img width="776" alt="Bildschirmfoto 2020-06-02 um 5 41 28 PM" src="https://user-images.githubusercontent.com/10846939/83540325-a9b78a00-a4e7-11ea-8cf6-3412abaae091.png">

Now, the code uses try-with-resource statements which is logically different, i just wanted to point this out. If you can give me an example how to show the stop method and using try-with-resource statements in the source code. Currently the docs would look like this:
<img width="776" alt="Bildschirmfoto 2020-06-02 um 5 41 40 PM" src="https://user-images.githubusercontent.com/10846939/83540246-90aed900-a4e7-11ea-8a8e-3f300fd8e85e.png">